packages/integration-tests/src/test-in-memory/integration_tests.test.ts

```diff
@@ -39,7 +39,7 @@ void it('data storage auth with triggers', () => {
 
   assertExpectedLogicalIds(templates.storage, 'AWS::S3::Bucket', [
     // eslint-disable-next-line spellcheck/spell-checker
-    'amplifyStorageamplifyStorageBucketC2F702CD',
+    'testNametestNameBucketABFD5AA0',
```

Comment:
Why is `testName` duplicated here?

Reply:
seems like we should drop `${id}` from bucket id.

Reply:
We can't do that because each construct id within an app must be unique. We could however completely remove the `AmplifyStorage` construct (or just remove the `extends Construct`) and place the bucket directly in the stack

Reply:
This seems to work.

```
import * as cdk from 'aws-cdk-lib';
import { Construct } from 'constructs';
import * as s3 from 'aws-cdk-lib/aws-s3';

class Construct1 extends Construct {
  constructor(scope: Construct) {
    super(scope, 'id1');
    new s3.Bucket(this, 'Bucket');
  }
}

class Construct2 extends Construct {
  constructor(scope: Construct) {
    super(scope, 'id2');
    new s3.Bucket(this, 'Bucket');
  }
}

export class Testapp005Stack extends cdk.Stack {
  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);

    new Construct1(this);
    new Construct2(this);
  }
}

```

Yields:

```
{
 "Resources": {
  "id1Bucket1929D7E3": {
   "Type": "AWS::S3::Bucket",
   "UpdateReplacePolicy": "Retain",
   "DeletionPolicy": "Retain",
   "Metadata": {
    "aws:cdk:path": "Testapp005Stack/id1/Bucket/Resource"
   }
  },
  "id2Bucket3331D0EF": {
   "Type": "AWS::S3::Bucket",
   "UpdateReplacePolicy": "Retain",
   "DeletionPolicy": "Retain",
   "Metadata": {
    "aws:cdk:path": "Testapp005Stack/id2/Bucket/Resource"
   }
  },
  "CDKMetadata": {
   "Type": "AWS::CDK::Metadata",
   "Properties": {
    "Analytics": "v2:deflate64:H4sIAAAAAAAA/zPSszTWM1BMLC/WTU7J1s3JTNKrDi5JTM7WAQrFFxvrVTuVJmenlug4p+VBWLUgZlBqcX5pUXIqiO2cn5eSWZKZn1erk5efkqqXVaxfZmihZ2iuZ6iYVZyZqVtUmleSmZuqFwShAZbAhltzAAAA"
   },
   "Metadata": {
    "aws:cdk:path": "Testapp005Stack/CDKMetadata/Default"
   },
   "Condition": "CDKMetadataAvailable"
  }
 },
 
...
```

Reply:
Oh interesting, I guess it only has to be unique within it's immediate scope. Will update

Reply:
Done